### PR TITLE
[Automation-8] Change validation to use staging Docker images: auth-based access, DSL update, DSL generator update

### PR DESCRIPTION
**Pull request by AndreyKoltsov1997**
**Now**
We access images from public Dockerhub repository.

**Then**
We must provide convenient API to access private Dockerhub repositories.

### CODE REVIEW

File: .teamcity/generated/ImageValidation.kts [hunk 1]
```diff
@@ -17,86 +17,79 @@ import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.finishBuildTrigger
...
 
-		triggers {
-			// Execute the build once the images are available within %deployRepository%
-			finishBuildTrigger {
-				buildType = "${PublishHubVersion.publish_hub_version.id}"
-			}
+	triggers {
+		// Execute the build once the images are available within %deployRepository%
+		finishBuildTrigger {
+			buildType = "${PublishHubVersion.publish_hub_version.id}"
```
Comment (burnasheva) [inline .teamcity/generated/ImageValidation.kts:30]: This is the wrong dependency (at this time images are published to the docker hub, and we want to check it on the staging). So, the right dependency would be the TC_Trunk_DockerImages_publish_local configuration.

File: tool/automation/framework/app/src/main/kotlin/com/jetbrains/teamcity/common/constants/ValidationConstants.kt [hunk 2]
```diff
@@ -5,4 +5,5 @@ package com.jetbrains.teamcity.common.constants
  */
 object ValidationConstants {
     const val ALLOWED_IMAGE_SIZE_INCREASE_THRESHOLD_PERCENT = 5.0f
+    const val PRE_PRODUCTION_IMAGE_PREFIX = "EAP"
```
Comment (burnasheva) [inline tool/automation/framework/app/src/main/kotlin/com/jetbrains/teamcity/common/constants/ValidationConstants.kt:8]: For which cases do we need it?

**Review by burnasheva — CHANGES_REQUESTED** [review]

File: tool/automation/framework/app/src/main/kotlin/com/jetbrains/teamcity/common/constants/ValidationConstants.kt (hunk 2 above)
Comment (AndreyKoltsov1997) [reply tool/automation/framework/app/src/main/kotlin/com/jetbrains/teamcity/common/constants/ValidationConstants.kt:8]: @burnasheva it was added in order to exclude new `EAP` images inside of staging registry from the comparison, having an assumption we don't treat it as a specific release. 
A related filtering is applied within [this line](https://github.com/JetBrains/teamcity-docker-images/pull/41/files#diff-159f861e7ed104470463497ea1d14cf82d0b4f070911f315148bfc53d73d5f8dR99).

Comment (burnasheva) [reply tool/automation/framework/app/src/main/kotlin/com/jetbrains/teamcity/common/constants/ValidationConstants.kt:8]: I see, thank you. We stopped publishing EAP tag several years ago, but if it's only for filtering I don't see anything bad in having it here.

**Review by burnasheva — APPROVED** [review]

**Pull request merged**